### PR TITLE
Bug Fix: `azurerm_app_service_plan` and `azurerm_app_service_slot` crash fixes

### PR DESCRIPTION
**Pull request by mbfrahry**

### CODE REVIEW

**Comment by mbfrahry** [comment]
Addressing #3760

**Comment by mbfrahry** [comment]
image

File: website/docs/r/app_service_slot.html.markdown [hunk 1]
```diff
@@ -306,7 +306,10 @@ A `google` block supports the following:
 
 A `ip_restriction` block supports the following:
 
-* `ip_address` - (Required) The IP Address used for this IP Restriction.
+* `ip_address` - (Optional) The IP Address used for this IP Restriction. One of `ip_address` or `virtual_network_subnet_id` must be specified.
+
+* `virtual_network_subnet_id` - (Optional.The Virtual Network Subnet ID used for this IP Restriction. One of `ip_address` or `virtual_network_subnet_id` must be specified.
```
Comment (tombuildsstuff) [inline website/docs/r/app_service_slot.html.markdown]: (same here)

**Comment by jeffadavidson** [comment]
In version 1.34.0 I am now getting an error that the virtual_network_subnet_id is required

`  on main.tf line 43, in resource "azurerm_app_service" "*******":
  43:     ip_restriction = [

Inappropriate value for attribute "ip_restriction": element 0: attribute
"virtual_network_subnet_id" is required.`

**Comment by mbfrahry** [comment]
Hi @jeffadavidson, do you mind opening an issue with additional information? Once I have that, I can take a crack at figuring out what's going on.

**Pull request merged**